x/subspaces/keeper/feegrant.go

```diff
@@ -87,6 +102,50 @@ func (k Keeper) GetGroupGrant(ctx sdk.Context, subspaceID uint64, groupID uint32
 
 // --------------------------------------------------------------------------------------------------------------------
 
+// saveAllowanceToExpiringQueue saves the allowance into expiring queue
+func (k Keeper) saveAllowanceToExpiringQueue(ctx sdk.Context, expiration *time.Time, grantKey []byte) {
+	k.removeAllowanceFromExpiringQueue(ctx, grantKey)
```

Comment:
Why is this needed?

Reply:
If a grant updated, we should make sure the grant with original expiration is removed first. 

For instance, there is a grant expired at `20220707` and we would like to update the expiration time to `20220708`, the grant will have two expiring key if it is not removed the old one first, like:
```
expiring key | 20220707 | grant key
expiring key | 20220708 | grant key
```

Reply:
Ok I see. I think it's better to add a comment then. Something like: 

```
// Make sure we remove the grant from the expiring queue to properly handle expiration date updates and avoid duplicated keys
```

Reply:
Added the comment